lib/nerves_network/dhcp_manager.ex

```diff
@@ -55,216 +37,223 @@ defmodule Nerves.Network.DHCPManager do
   # :is_lower_up reports whether the interface as associated with an AP
-  defp handle_registry_event({Nerves.NetworkInterface, :ifchanged, %{ifname: ifname, is_lower_up: true}}) do
-    Logger.debug "DHCPManager(#{ifname}) network_interface ifup"
+  def handle_event({Nerves.NetworkInterface, :ifchanged, %{ifname: ifname, is_lower_up: true}}) do
+    Logger.debug "DHCPManager.EventHandler(#{ifname}) ifup"
     :ifup
   end
-
-  defp handle_registry_event({Nerves.NetworkInterface, :ifchanged, %{ifname: ifname, is_lower_up: false}}) do
-    Logger.debug "DHCPManager(#{ifname}) network_interface ifdown"
+  def handle_event({Nerves.NetworkInterface, :ifchanged, %{ifname: ifname, is_lower_up: false}}) do
+    Logger.debug "DHCPManager.EventHandler(#{ifname}) ifdown"
     :ifdown
   end
 
-  defp handle_registry_event({Nerves.NetworkInterface, event, %{ifname: ifname}}) do
-    Logger.debug "DHCPManager(#{ifname}): ignoring event: #{inspect event}"
-    :noop
-  end
-
-  defp handle_registry_event({Nerves.Udhcpc, :bound, %{ifname: ifname} = info}) do
-    Logger.info "WiFiManager(#{ifname}): udhcp bound"
-    {:bound, info}
-  end
+  # # DHCP events
+  # # :bound, :renew, :rebind, :nak
 
-  defp handle_registry_event({Nerves.Udhcpc, :deconfig, %{ifname: ifname} = info}) do
-    Logger.info "WiFiManager(#{ifname}): udhcp deconfig"
-    {:deconfig, info}
-  end
-
-  defp handle_registry_event({Nerves.Udhcpc, :nak, %{ifname: ifname} = info}) do
-    Logger.info "WiFiManager(#{ifname}): udhcp nak"
-    {:nak, info}
-  end
 
-  defp handle_registry_event({Nerves.Udhcpc, :renew, %{ifname: ifname} = info}) do
-    Logger.info "WiFiManager(#{ifname}): udhcp renew"
-    {:renew, info}
-  end
-
-  defp handle_registry_event({Nerves.Udhcpc, event, %{ifname: ifname}}) do
-    Logger.info "WiFiManager(#{ifname}): ignoring event: #{inspect event}"
+  def handle_event({Nerves.NetworkInterface, event, %{ifname: ifname}}) do
+    Logger.debug "DHCPManager.EventHandler(#{ifname}): ignoring event: #{inspect event}"
     :noop
   end
 
-  # Handle Network Interface events coming in from SystemRegistry.
   def handle_info({Nerves.NetworkInterface, _, ifstate} = event, %{ifname: ifname} = s) do
-    event = handle_registry_event(event)
+    Logger.debug fn -> "#{__MODULE__} handle_info: ifstate = #{inspect ifstate}" end
+    event = handle_event(event)
     scope(ifname) |> SystemRegistry.update(ifstate)
     s = consume(s.context, event, s)
-    Logger.debug "DHCPManager(#{s.ifname}, #{s.context}) got event #{inspect event}"
+    Logger.debug fn -> "DHCPManager(#{s.ifname}, #{s.context}) got event #{inspect event}" end
     {:noreply, s}
   end
 
-  # Handle Udhcpc events coming from SystemRegistry.
-  def handle_info({Nerves.Udhcpc, event, info}, %{ifname: ifname} = s) do
-    Logger.debug "DHCPManager(#{s.ifname}) udhcpc #{inspect event}"
+  #  info: %{domain_search: "ipv4.doman.name", ifname: "eth1", ipv4_address: "666::16/64", nameservers: ["fec0:0:0:1::7"]}
```

Comment:
the comment above is from ipv6.. we may want to either alter it to reflect the ipv4 reality or remove it